appcli/commands/configure_cli.py

```diff
@@ -171,16 +205,22 @@ def __seed_configuration_dir(self, cli_context: CliContext):
                 logger.debug("Copying seed file to [%s] ...", target_file)
                 shutil.copy2(source_file, target_file)
 
+        # After initialising the configuration directory, put it under source control
+        ConfigurationGitRepository(cli_context).init()
+
     def __generate_configuration_files(
         self, configuration: Configuration, cli_context: CliContext
     ):
         self.__print_header(f"Generating configuration files")
         generated_configuration_dir = cli_context.generated_configuration_dir
+
+        # If the generated configuration directory is not empty, back it up and delete
+        if os.listdir(generated_configuration_dir):
+            self._backup_and_remove_directory(generated_configuration_dir)
+
         generated_configuration_dir.mkdir(parents=True, exist_ok=True)
 
-        configuration_record_file = generated_configuration_dir.joinpath(
-            METADATA_FILE_NAME
-        )
+        configuration_record_file = get_metadata_file_directory(cli_context)
```

Comment:
Moved this functionality to the `functions.py` so it can also be used by `MainCli` when getting the commit #